typings/entry.d.ts

```diff
@@ -18,20 +18,20 @@ export interface EntrySys extends MetaSysProps {
   publishedCounter?: number,
 }
 
-export interface Entry
-  extends EntryProp,
+export interface Entry<TEntryFields = any>
+  extends EntryProp<TEntryFields>,
     DefaultElements<EntryProp & EntrySys>,
     MetaSys<EntrySys> {
-  archive(): Promise<Entry>,
+  archive<TFieldType = {}>(): Promise<Entry<TFieldType>>,
   delete(): Promise<void>,
   getSnapshot(id: string): Promise<any>,
   getSnapshots(): Promise<Collection<any>>,
   isArchived(): boolean,
   isDraft(): boolean,
   isPublished(): boolean,
   isUpdated(): boolean,
-  publish(): Promise<Entry>,
-  unarchive(): Promise<Entry>,
-  unpublish(): Promise<Entry>,
-  update(): Promise<Entry>,
+  publish<TFieldType = {}>(): Promise<Entry<TFieldType>>,
```

Comment:
I think the `publish` method returns the Entry based on `TEntryFields`. So, we can take out the generic template which defaults to an empty object.

You can reference my previous comment on the Entry interface for what this would look like.